### PR TITLE
Group Preferences Actions

### PR DESCRIPTION
**Pull request by zombieobject**
## Introduction 📟
This Pull Request is intended to add Consent/Deny functionality to actions related to Group Messaging. 

## Discussion 🎙
The implementation follows closely with the work completed in the following [Android SDK Pull Request](https://github.com/xmtp/xmtp-android/pull/180)

### CODE REVIEW

**Review by neekolas — COMMENTED** [review]
Not enough of an xmtp-ios expert to approve, but love to see this coming to life

File: Sources/XMTPiOS/Proto/message_contents/message.pb.swift [hunk 1]
```diff
@@ -125,16 +125,32 @@ public struct Xmtp_MessageContents_MessageV2 {
...
   /// HMAC of the message ciphertext, with the HMAC key derived from the topic key
-  public var senderHmac: Data = Data()
+  public var senderHmac: Data {
+	get {return _senderHmac ?? Data()}
+	set {_senderHmac = newValue}
+  }
+  /// Returns true if `senderHmac` has been explicitly set.
+  public var hasSenderHmac: Bool {return self._senderHmac != nil}
+  /// Clears the value of `senderHmac`. Subsequent reads from it will return its default value.
+  public mutating func clearSenderHmac() {self._senderHmac = nil}
```
Comment (tuddman) [inline Sources/XMTPiOS/Proto/message_contents/message.pb.swift:135]: Is this related to this PR?

**Review by tuddman — APPROVED** [review]
One ? about the inclusion of `hmac` functionality.  Otherwise it looks very similar and good!

**Comment by nplasterer** [comment]
Sorry for the late review on this. But looks like you missed updating one place 
https://github.com/xmtp/xmtp-ios/blob/main/Sources/XMTPiOS/Conversation.swift#L33-L46

This whole method needs to be rewritten to better align with Android. https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/Conversation.kt#L91-L97
